R/pool-methods.R

```diff
@@ -0,0 +1,89 @@
+#' @include pool.R
+NULL
+
+#' S4 class for compatibility with DBI methods
+#' @export
+setClass("Pool")
+
+## documented manually, with the Pool object
+#' @export
+poolCreate <- function(factory, ...,
+                       minSize = 1, maxSize = Inf,
+                       idleTimeout = 60000, validateTimeout = 2000,
+                       stateEnv = NULL) {
+  pool <- Pool$new(
+    function() {factory(...)},
+    minSize, maxSize,
+    idleTimeout, validateTimeout, stateEnv
+  )
+  ## if a createPool is used with a DBIDriver, make a note
+  ## (to ease dplyr compatibility later on)
+  checkDriver(pool, ...)
```

Comment:
1) Yes, I was just curious. It does make sense to be consistent at that level (I was just wondering if there was some other technical reason we'd want to do this). [I also changed the name of `stateEnv` back to the neater `state` -- I had it as `stateEnv` before just not to confuse it with the object's state ("free" or "taken"). But the `state` argument is only used in the `initialize` function (in which no object's state is relevant). From then on, we always need to do `self$state`, so I figured the same name wasn't a real problem.]

2. Ok, that makes sense. I had still been seeing `createPool` and `dbPool` as kind of equivalents up to now. But there's really no reason that should be the case...